### PR TITLE
infoschema: refactor GetClusterServerInfo to avoid executing internal SQL

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

In the current implementation, we must to execute `SELECT type, address, status_address FROM INFORMATION_SCHEMA.TIDB_CLUSTER_INFO ORDER BY type` to obtain cluster topology. That's inefficient.

### What is changed and how it works?

This PR refactor this part and extras an individual function to obtains cluster topology and make other functions work more easily.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Release note

 - No need.

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -1909,25 +1908,51 @@ func dataForServersInfo() ([][]types.Datum, error) {
...
+type ServerInfo struct {
+	ServerType string
+	Address    string
+	StatusAddr string
+	Version    string
+	GitHash    string
+}
+
+// GetClusterServerInfo returns all components information of cluster
+func GetClusterServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
```
Comment (crazycs520) [inline infoschema/tables.go]: No need export this function?

File: go.mod [hunk 2]
```diff
@@ -2,7 +2,6 @@ module github.com/pingcap/tidb
 
 require (
 	github.com/BurntSushi/toml v0.3.1
-	github.com/StackExchange/wmi v0.0.0-20180725035823-b12b22c5341f // indirect
```
Comment (crazycs520) [inline go.mod]: No need change go.sum, go.mod ?

File: infoschema/tables.go (hunk 1 above)
Comment (lonng) [reply infoschema/tables.go]: The` perfschema` will reuse this function.

File: go.mod (hunk 2 above)
Comment (lonng) [reply go.mod]: Use `go mod tidy` to make CI happy.

**Comment by djshow832** [comment]
Change the PR title: inforschema => infoschema, execute => executing.

**Comment by lonng** [comment]
@djshow832 Please approve this PR.

**Pull request merged**